hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,256 @@
+
+In response to the low power entry request, the fast FSM disables all second level clock gating and asserts appropriate resets if required.  The
+fast FSM then requests the slow FSM to take over.
+Once the request is acknowledged, the fast FSM transitions to low power and waits for the next power up request.
+
+
+## Reset Request Handling
+
+There are 3 reset requests in the system - peripheral requested reset such as watchdog, alert escalation reset and non-debug module reset.
+Flash brownout is handled separately and described in [flash handling section](#Flash-Handling) below.
+
+Watchdog and alert escalation resets are handled directly by the power manager, while the non-debug module reset is handled by the reset controller.
+This separation is because the non-debug reset does not affect the life cycle controller, non-volatile storage controllers and alert states.
+There is thus no need to sequence its operation like the others.
+
+The power controller only observes reset requests in two states - the slow FSM `Low_power` state and the fast FSM `Active` state.
+When a reset request is received during slow FSM `Low_power` state, the system begins its usual power up sequence even if a wakeup has not been received.
+
+When a reset request is received during fast FSM `Active` state, the fast FSM asserts resets and transitions back to its `Low_power` state.
+The normal power-up process described [above](#Fast-Clock-Domain-FSM) is then followed to release the resets.
+Note in this case, the slow FSM is “not activated” and remains in its `Idle` state.
+
+
+### Reset Requests Received During Other States
+
+All other states in the slow / fast FSM are considered transitional states.
+Resets are not observed in other states because the system will always be transitioning towards one of the steady states (the system is in the process of powering down or powering up).
+Once a steady state is reached, reset requests are then observed and processed.
+
+### Reset Recording
+
+There are three ways in which the device is reset:
+- Non-debug-module reset request
+- Low power entry
+- Direct reset requests by peripherals or alert escalation
+
+The power manager does not handle the non-debug-module request (please see reset controller).
+For the remaining two reset causes, the power manager handles only 1 pathway at a time (see state diagrams).
+This means if reset request and low power entry collide, the power manager will handle on a first come first serve basis.
+When the handling of the first is completed, the power manager handles the second pending request if it is still present.
+
+This is done because low power resets and peripheral requested resets lead to different behaviors.
+When the power manager commits to handling a specific request, it informs the reset manager why it has reset the processor.
+
+For example, assume low power entry request arrives slightly ahead of reset requests, the power manager will:
```

Comment:
I think this needs an article "assume **a** low power entry request...". Also (nit), you probably want to break the sentence to give something like "... reset requests. Then the power manager will:". (Otherwise, the comma separates two primary clauses, which ain't allowed)

Reply:
fixed